clients/vscode/src/commands/commandPalette.ts

```diff
@@ -16,60 +19,85 @@ export class CommandPalette {
   show() {
     const quickPick: QuickPick<CommandPaletteItem> = window.createQuickPick();
     quickPick.title = "Tabby Command Palette";
+    let items: CommandPaletteItem[] = [];
 
-    const items: CommandPaletteItem[] = [];
+    const disconnectStatues = ["disconnected", "unauthorized"];
+    if (disconnectStatues.includes(this.client.status.current?.status || "")) {
+      items.push(this.itemForStatus());
+    }
 
-    items.push(this.itemForStatus());
+    // Status section
     this.client.status.on("didChange", () => {
       items[0] = this.itemForStatus();
       quickPick.items = items;
     });
 
+    // Features section
+    const validStatuses = ["ready", "readyForAutoTrigger", "readyForManualTrigger"];
+    if (validStatuses.includes(this.client.status.current?.status || "")) {
+      items = items.concat([
+        { label: "enable/disable features", kind: QuickPickItemKind.Separator },
+        {
+          label:
+            (this.config.inlineCompletionTriggerMode === "automatic" ? "" : MENU_ITEM_INDENT_SPACING) +
+            "Code Completion",
+          detail: MENU_ITEM_INDENT_SPACING + "Toggle between automatic and manual completion mode",
+          picked: this.config.inlineCompletionTriggerMode === "automatic",
+          command: "tabby.toggleInlineCompletionTriggerMode",
+          iconPath: this.config.inlineCompletionTriggerMode === "automatic" ? new ThemeIcon("check") : undefined,
+          alwaysShow: true,
+        },
+      ]);
+    }
+
+    // Chat section
     if (this.client.chat.isAvailable) {
       items.push({
         label: "$(comment) Chat",
         command: "tabby.chatView.focus",
       });
     }
 
-    items.push(
-      {
-        label: "",
-        kind: QuickPickItemKind.Separator,
-      },
-      {
-        label: this.config.inlineCompletionTriggerMode === "manual" ? "Enable Completions" : "Disable Completions",
-        command: "tabby.toggleInlineCompletionTriggerMode",
-      },
+    // Settings section
+    items = items.concat([
+      { label: "settings", kind: QuickPickItemKind.Separator },
       {
-        label: "",
-        kind: QuickPickItemKind.Separator,
-      },
-      {
-        label: "$(plug) Connect to Server...",
+        label: "Connect to Server",
         command: "tabby.connectToServer",
+        iconPath: new ThemeIcon("plug"),
       },
       {
-        label: "$(settings) Settings...",
+        label: "Settings",
         command: "tabby.openSettings",
+        iconPath: new ThemeIcon("gear"),
```

Comment:
I prefer use icon `settings` than `gear` as VSCode settings page use this icon.

![20241212-194752](https://github.com/user-attachments/assets/949e7028-84c5-4031-afca-27ed1f8ece36)